exporter/dynatraceexporter/internal/serialization/serialization_test.go

```diff
@@ -0,0 +1,179 @@
+	staticDims := dimensions.NewNormalizedDimensionList(dimensions.NewDimension("static", "value"))
+
+	t.Run("correctly creates a gauge", func(t *testing.T) {
+		metric := pmetric.NewMetric()
+		metric.SetDataType(pmetric.MetricDataTypeGauge)
+		metric.SetName("metric_name")
+		gaugeDp := metric.Gauge().DataPoints().AppendEmpty()
+		gaugeDp.SetIntVal(3)
+
+		prev := ttlmap.New(1, 1)
+
+		serialized, err := SerializeMetric(logger, "prefix", metric, defaultDims, staticDims, prev)
+		assert.NoError(t, err)
+
+		assert.Len(t, serialized, 1)
+
+		assertMetricLineTokensEqual(t, serialized[0], "prefix.metric_name,default=value,static=value gauge,3")
+	})
+
+	t.Run("correctly creates a counter from a sum", func(t *testing.T) {
+		// more in-depth tests for the different sum serializations are in sum_test.go
+		metric := pmetric.NewMetric()
+		metric.SetDataType(pmetric.MetricDataTypeSum)
+		metric.SetName("metric_name")
+		sum := metric.Sum()
+		sum.SetIsMonotonic(true)
+		sum.SetAggregationTemporality(pmetric.MetricAggregationTemporalityDelta)
+		sumDp := sum.DataPoints().AppendEmpty()
+		sumDp.SetIntVal(4)
+
+		prev := ttlmap.New(1, 1)
+
+		serialized, err := SerializeMetric(logger, "prefix", metric, defaultDims, staticDims, prev)
+		assert.NoError(t, err)
+
+		assert.Len(t, serialized, 1)
+
+		assertMetricLineTokensEqual(t, serialized[0], "prefix.metric_name,default=value,static=value count,delta=4")
+	})
+
+	t.Run("correctly creates a summary from histogram", func(t *testing.T) {
+		metric := pmetric.NewMetric()
+		metric.SetDataType(pmetric.MetricDataTypeHistogram)
+		metric.SetName("metric_name")
+		hist := metric.Histogram()
+		hist.SetAggregationTemporality(pmetric.MetricAggregationTemporalityDelta)
+		dp := hist.DataPoints().AppendEmpty()
+		dp.SetMin(1)
+		dp.SetMax(3)
+		dp.SetSum(6)
+		dp.SetCount(3)
+
+		prev := ttlmap.New(1, 1)
+
+		serialized, err := SerializeMetric(logger, "prefix", metric, defaultDims, staticDims, prev)
+		assert.NoError(t, err)
+
+		assert.Len(t, serialized, 1)
+
+		assertMetricLineTokensEqual(t, serialized[0], "prefix.metric_name,default=value,static=value gauge,min=1,max=3,sum=6,count=3")
+	})
+
+}
+
+func Test_makeCombinedDimensions(t *testing.T) {
+	defaultDims := dimensions.NewNormalizedDimensionList(
+		dimensions.NewDimension("a", "default"),
+		dimensions.NewDimension("b", "default"),
+		dimensions.NewDimension("c", "default"),
+	)
+	labels := pcommon.NewMap()
+	labels.Insert("a", pcommon.NewValueString("label"))
+	labels.Insert("b", pcommon.NewValueString("label"))
+	staticDims := dimensions.NewNormalizedDimensionList(
+		dimensions.NewDimension("a", "static"),
+	)
+	expected := dimensions.NewNormalizedDimensionList(
+		dimensions.NewDimension("a", "static"),
+		dimensions.NewDimension("b", "label"),
+		dimensions.NewDimension("c", "default"),
+	)
+
+	actual := makeCombinedDimensions(labels, defaultDims, staticDims)
```

Comment:
Should we arrange the test in a way such as `staticDims, defaultDims, labels` are passed in this order to make sure datapoint dimensions override static/default ones? 🤔

Reply:
changed it in 52ae92c1. Leaving this conversation unresolved, is anyone strongly against this change?